src/adapter/src/coord/command_handler.rs

```diff
@@ -911,8 +911,15 @@ impl Coordinator {
                 maybe_ctx = Some(ctx);
             }
 
+            // Cancel reads waiting on being linearized. There is at most one linearized read per
+            // session.
+            if let Some(pending_read_txn) = self.pending_linearize_read_txns.remove(&conn_id) {
+                let ctx = pending_read_txn.take_context();
+                maybe_ctx = Some(ctx);
+            }
+
             if let Some(ctx) = maybe_ctx {
-                ctx.retire(Ok(ExecuteResponse::Canceled));
+                ctx.retire(Err(AdapterError::Canceled));
```

Comment:
This IMO is the biggest difference. We are able to respond both with an `Ok(ExecuteResponse::Canceled)` and an `Err(AdapterError::Canceled)`. They have different semantics with respect to what the client ultimately receives that I don't think I full understand, however I think sending an `Err` here is the right approach. I'm going to post a comment on a specific test explaining why.